yarn-project/end-to-end/src/e2e_call_contract_same_block.test.ts

```diff
@@ -0,0 +1,46 @@
+import { ContractFunctionInteraction, Wallet } from '@aztec/aztec.js';
+import { TokenContract } from '@aztec/noir-contracts/types';
+import { TxStatus } from '@aztec/types';
+
+import times from 'lodash.times';
+
+import { setup } from './fixtures/utils.js';
+
+describe('e2e_call_contract_same_block', () => {
+  let owner: Wallet;
+  let minter: Wallet;
+  let teardown: () => Promise<void>;
+
+  describe('multi-txs block', () => {
+    beforeAll(async () => {
+      ({
+        teardown,
+        wallets: [owner, minter],
+      } = await setup(2));
+    }, 100_000);
+
+    afterAll(() => teardown());
+
+    it('can call public function from different tx in same block', async () => {
+      // Deploy a contract in the first transaction
+      // In the same block, call a public method on the contract
+      const deployer = TokenContract.deploy(owner, owner.getCompleteAddress());
+      await deployer.create();
+
+      const callInteraction = new ContractFunctionInteraction(
```

Comment:
Are you not able to call the TokenContract.at() function to get an instance of generated interface on which to call?